File: generate-10-posts.js
// Generate 10 blog posts with diverse Colemearchy themes
const blogTopics = [
  {
    prompt: "AI 시대에도 살아남는 개발자가 되는 법. 10년차 개발자가 말하는 AI와 공존하며 성장하는 전략. ChatGPT, Claude, Cursor를 활용한 실전 사례와 차별화 방법",
    keywords: ["AI 시대", "개발자 생존", "ChatGPT", "Claude", "Cursor", "개발자 차별화"]
  },
  {
    prompt: "ADHD 개발자의 극한 집중력 해킹법. 산만함을 슈퍼파워로 바꾼 개인적 경험과 과학적 근거. 포모도로, 노이즈 캔슬링, 약물치료까지",
    keywords: ["ADHD", "프로그래밍", "집중력", "생산성", "포모도로", "노이즈캔슬링"]
  },
  {
    prompt: "개발자를 위한 FIRE 운동 실전 가이드. 연봉 1억 개발자가 경제적 자유를 달성한 투자 전략. 주식, 부동산, 암호화폐 포트폴리오 공개",
    keywords: ["FIRE", "개발자 투자", "경제적 자유", "주식", "부동산", "암호화폐"]
  },
  {
    prompt: "개발자 생산성 200% 올리는 바이오해킹 전략. 수면 최적화, 영양제 스택, 간헐적 단식부터 콜드 샤워까지. 6개월 실험 결과 공개",
    keywords: ["바이오해킹", "개발자 생산성", "수면 최적화", "영양제", "간헐적 단식"]
  },
  {
    prompt: "스타트업 CTO가 되기까지의 여정과 리더십 철학. 실패한 3개의 스타트업에서 배운 교훈과 성공하는 기술 조직 만들기",
    keywords: ["스타트업 CTO", "리더십", "기술 조직", "스타트업 실패", "팀 빌딩"]
  },
  {
    prompt: "ChatGPT vs Claude vs Gemini 개발자를 위한 실전 비교. 코드 리뷰, 아키텍처 설계, 디버깅까지 6개월 사용 후기와 최적 활용법",
    keywords: ["ChatGPT", "Claude", "Gemini", "AI 도구", "개발 생산성"]
  },
  {
    prompt: "코딩으로 망가진 목과 허리 되살리기. 거북목과 허리 디스크에서 탈출한 개발자의 3개월 재활기. 스탠딩 데스크부터 요가까지",
    keywords: ["거북목", "허리 디스크", "개발자 건강", "스탠딩 데스크", "요가"]
  },
  {
    prompt: "위고비에서 케토까지, 개발자의 체중 감량 실험기. 3개월간 15kg 감량하며 배운 것들. 부작용과 리바운드 극복법",
    keywords: ["위고비", "케토", "체중 감량", "개발자 다이어트", "간헐적 단식"]
  },
  {
    prompt: "리모트 워크 3년, 낭만과 현실 사이. 디지털 노마드의 꿈과 실제 경험한 장단점. 생산성, 커뮤니케이션, 워라밸의 진실",
    keywords: ["리모트 워크", "재택근무", "디지털 노마드", "워라밸", "생산성"]
  },
  {
    prompt: "부트캠프 vs CS 학위, 10년차 개발자가 말하는 진실. 실무에서 정말 중요한 것과 학력의 한계. 채용 담당자의 속마음",
    keywords: ["부트캠프", "CS 학위", "개발자 교육", "채용", "실무 경험"]
  }
];

async function generateBlogPost(topic, index) {
  try {
    console.log(`\n[${index + 1}/${blogTopics.length}] Generating post...`);
    console.log(`Topic: ${topic.prompt.substring(0, 50)}...`);
    
    const response = await fetch('http://localhost:3001/api/generate-content', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        prompt: `colemearchy 스타일로 ${topic.prompt}에 대한 깊이 있는 블로그 포스트 작성`,
        keywords: topic.keywords,
        type: 'blog-post'
      })
    });

    const data = await response.json();
    
    if (data.id) {
      // Add timestamp to ensure unique slug
      const timestamp = Date.now() + Math.floor(Math.random() * 10000);
      const uniqueSlug = data.slug + '-' + timestamp;
      
      // Publish the post
      const publishResponse = await fetch('http://localhost:3001/api/posts', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          ...data,
          slug: uniqueSlug,
          status: 'PUBLISHED',
          originalLanguage: 'ko'
        })
      });
      
      const publishedPost = await publishResponse.json();
      
      if (publishedPost.id) {
        console.log(`✅ Success: ${data.title}`);
        console.log(`   URL: http://localhost:3001/ko/posts/${publishedPost.slug}`);
        return { success: true, ...publishedPost };
      }
    }
    
    console.error(`❌ Failed to generate post`);
    return { success: false, error: 'Generation failed' };
  } catch (error) {
    console.error(`❌ Error:`, error.message);
    return { success: false, error: error.message };
  }
}

async function main() {
  console.log('📝 Generating 10 Colemearchy-style blog posts...');
  
  const results = {
    successful: [],
    failed: []
  };
  
  for (let i = 0; i < blogTopics.length; i++) {
    const result = await generateBlogPost(blogTopics[i], i);
    
    if (result.success) {
      results.successful.push(result);
    } else {
      results.failed.push({ ...blogTopics[i], error: result.error });
    }
    
    // Wait between posts
    if (i < blogTopics.length - 1) {
      console.log('⏳ Waiting 5 seconds...');
      await new Promise(resolve => setTimeout(resolve, 5000));
    }
  }
  
  console.log('\n📊 Generation Complete!');
  console.log(`✅ Successful: ${results.successful.length}`);
  console.log(`❌ Failed: ${results.failed.length}`);
  
  if (results.successful.length > 0) {
    console.log('\nSuccessfully created posts:');
    results.successful.forEach(s => {
      console.log(`- ${s.title}`);
    });
  }
  
  if (results.failed.length > 0) {
    console.log('\nFailed posts:');
    results.failed.forEach(f => {
      console.log(`- ${f.prompt.substring(0, 50)}...`);
    });
  }
}

main().catch(console.error);

File: generate-10-posts.py
#!/usr/bin/env python3
import requests
import json
import time
from datetime import datetime, timedelta

# Generate 10 NEW blog posts with diverse Colemearchy themes
new_blog_topics = [
    {
        "prompt": "생성형 AI로 부업 월 1000만원 벌기. ChatGPT, Claude, Midjourney를 활용한 실전 수익화 전략. 콘텐츠 제작, 자동화, SaaS까지",
        "keywords": ["AI 부업", "ChatGPT 수익화", "Claude 활용법", "Midjourney", "패시브 인컴", "자동화"]
    },
    {
        "prompt": "개발자의 번아웃 극복기. 3개월 휴직 후 찾은 진짜 내 삶. 일과 삶의 균형, 의미 있는 커리어 재정의",
        "keywords": ["번아웃", "개발자 휴직", "워라밸", "커리어 전환", "정신건강", "삶의 의미"]
    },
    {
        "prompt": "노마드 개발자 3년차의 리얼 비용 공개. 발리부터 리스본까지, 월 생활비와 세금 최적화 전략",
        "keywords": ["노마드", "디지털 노마드", "해외 거주", "생활비", "세금 최적화", "리모트 워크"]
    },
    {
        "prompt": "개발자가 만든 AI 자동화 홈오피스. 음성 비서부터 스마트 조명까지, 집중력 200% 올리는 환경 구축기",
        "keywords": ["스마트홈", "홈오피스", "자동화", "IoT", "생산성", "재택근무 환경"]
    },
    {
        "prompt": "FAANG 퇴사 후 1인 기업가의 첫해 수익 공개. 실패한 프로젝트와 성공한 SaaS, 그리고 배운 교훈들",
        "keywords": ["FAANG 퇴사", "1인 기업", "SaaS", "스타트업", "창업", "수익 공개"]
    },
    {
        "prompt": "AI 시대의 투자 포트폴리오 재구성. 테크 주식, AI ETF, 암호화폐 비중과 리밸런싱 전략",
        "keywords": ["AI 투자", "포트폴리오", "테크 주식", "ETF", "암호화폐", "자산 배분"]
    },
    {
        "prompt": "개발자를 위한 미니멀 라이프. 불필요한 것 버리고 본질에 집중하기. 디지털 디톡스부터 물건 정리까지",
        "keywords": ["미니멀리즘", "디지털 디톡스", "정리", "집중력", "본질", "단순한 삶"]
    },
    {
        "prompt": "뇌과학으로 해킹하는 개발자 학습법. 도파민 시스템 이해하고 활용하기. 습관 형성부터 장기 기억까지",
        "keywords": ["뇌과학", "학습법", "도파민", "습관", "기억력", "개발 공부"]
    },
    {
        "prompt": "개발자의 부동산 투자 실전기. REITs부터 직접 투자까지, 안정적 패시브 인컴 구축 전략",
        "keywords": ["부동산 투자", "REITs", "패시브 인컴", "개발자 재테크", "자산 형성", "임대 수익"]
    },
    {
        "prompt": "AI와 함께하는 개인 브랜딩. LinkedIn, Twitter, 개인 블로그로 영향력 있는 개발자 되기",
        "keywords": ["개인 브랜딩", "LinkedIn", "Twitter", "개발자 블로그", "영향력", "네트워킹"]
    }
]

def generate_blog_post(topic, index):
    try:
        print(f"\n[{index + 1}/{len(new_blog_topics)}] Generating post...")
        print(f"Topic: {topic['prompt'][:50]}...")
        
        # Calculate publish date (schedule posts 2 hours apart starting from next hour)
        now = datetime.now()
        publish_date = now + timedelta(hours=(index + 1) * 2)
        
        # Try production URL first
        urls = [
            'https://colemearchy.com/api/generate-content',
            'http://localhost:3000/api/generate-content',
            'http://localhost:3001/api/generate-content'
        ]
        
        data = None
        for url in urls:
            try:
                response = requests.post(url, 
                    json={
                        "prompt": f"colemearchy 스타일로 {topic['prompt']}에 대한 깊이 있는 블로그 포스트 작성. 개인적 경험과 실전 노하우를 담아서",
                        "keywords": topic['keywords'],
                        "publishDate": publish_date.isoformat()
                    },
                    timeout=120
                )
                
                if response.status_code == 200:
                    data = response.json()
                    print(f"✅ Using endpoint: {url}")
                    break
                else:
                    print(f"❌ {url} returned {response.status_code}")
            except Exception as e:
                print(f"❌ {url} failed: {str(e)}")
                continue
        
        if data and data.get('id'):
            print(f"✅ Success: {data.get('title', 'Untitled')}")
            print(f"   Status: {data.get('status', 'Unknown')}")
            print(f"   Scheduled for: {publish_date.strftime('%Y-%m-%d %H:%M:%S')}")
            print(f"   URL: https://colemearchy.com/posts/{data.get('slug', '')}")
            return {"success": True, **data}
        
        error_msg = data.get('error', 'Generation failed') if data else 'All endpoints failed'
        print(f"❌ Failed to generate post: {error_msg}")
        return {"success": False, "error": error_msg}
        
    except Exception as e:
        print(f"❌ Error: {str(e)}")
        return {"success": False, "error": str(e)}

def main():
    print('📝 Generating 10 NEW Colemearchy-style blog posts...')
    print('📅 Posts will be scheduled to publish every 2 hours starting from next hour')
    
    results = {
        "successful": [],
        "failed": []
    }
    
    for i, topic in enumerate(new_blog_topics):
        result = generate_blog_post(topic, i)
        
        if result.get("success"):
            results["successful"].append(result)
        else:
            results["failed"].append({**topic, "error": result.get("error", "Unknown error")})
        
        # Wait between posts to avoid rate limiting
        if i < len(new_blog_topics) - 1:
            print('⏳ Waiting 5 seconds...')
            time.sleep(5)
    
    print(f'\n📊 Generation Complete!')
    print(f'✅ Successful: {len(results["successful"])}')
    print(f'❌ Failed: {len(results["failed"])}')
    
    if results["successful"]:
        print('\nSuccessfully created posts:')
        for s in results["successful"]:
            print(f'- {s.get("title", "Untitled")}')
            if s.get("scheduledAt"):
                scheduled_time = datetime.fromisoformat(s["scheduledAt"].replace('Z', '+00:00'))
                print(f'  Scheduled: {scheduled_time.strftime("%Y-%m-%d %H:%M:%S")}')
    
    if results["failed"]:
        print('\nFailed posts:')
        for f in results["failed"]:
            print(f'- {f["prompt"][:50]}...')
            print(f'  Error: {f["error"]}')

if __name__ == "__main__":
    main()